project_task_pull_request_state/models/project_task.py

```diff
@@ -0,0 +1,108 @@
+# Copyright Cetmix OU 2023
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0).
+from odoo import api, fields, models
+
+
+class ProjectTaskState(models.Model):
+    _inherit = "project.task"
+
+    pr_state = fields.Selection(
+        selection=lambda self: self._selection_pr_state(),
+        tracking=True,
+        copy=False,
+        string="PR State",
+        compute="_compute_pr_state",
+        readonly=False,
+    )
+
+    def _selection_pr_state(self):
+        """Function to select the state of the pull request"""
+        return [
+            ("open", "Open"),
+            ("draft", "Draft"),
+            ("merged", "Merged"),
+            ("closed", "Closed"),
+        ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_recs = super(ProjectTaskState, self).create(vals_list)
+
+        # Set pr state if URI is provided at creation
+        for new_rec, vals in zip(new_recs, vals_list):
+            if "pr_uri" in vals:
+                new_rec._set_pr_state(vals.get("pr_uri"))
+        return new_recs
+
+    def write(self, vals):
+        if "pr_uri" in vals:
+            self._set_pr_state(vals.get("pr_uri"), vals.get("pr_state"))
+        return super(ProjectTaskState, self).write(vals)
+
+    @api.depends("pr_uri", "project_id.pr_state_default")
```

Comment:
If someone changes "pr_state_default" in project settings this might lead to all the project tasks PR state being updated. Even if it was set manually before. Do we really need this?

Reply:
resolved